### PR TITLE
fix(cassandra-harry): install from upstream master branch

### PR DESCRIPTION
**Pull request by fruch**
Now that https://github.com/apache/cassandra-harry/pull/12 is
merged, we can install and run from master branch.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by roydahan** [comment]
@fruch did you give it a try?

**Comment by fruch** [comment]
> @fruch did you give it a try?

Not yet

**Comment by fruch** [comment]
@roydahan 

Installation now works, but after ~1h of run:
https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-harry-2h-test/5/

it's failing on verification
```
Caused by: java.util.concurrent.ExecutionException: harry.model.Model$ValidationException: Found a row in the model that is not present in the resultset:
Expected: RowState{cd=2445118067971306811, vds=[506971169, 37516, 3690524376, 14369007], lts=[1499989, 1499989, 1499989, 1499989], clustering=[695496207763, 7.503688E-39], values=[506971169, -28020, -604442920, 2.0135267E-38]}
Actual: resultSetRow(6949810855733628783L, 3100022260925885995L, values(52145469098742527L,176L,40351L,168L,113L), lts(1499999L,1499999L,1499999L,1499999L,1499999L), values(3734862285L,-9223372036854775808L,-9223372036854775808L,-9223372036854775808L), lts(1499559L,-9223372036854775808L,-9223372036854775808L,-9223372036854775808L))
Query: CompiledStatement{cql='SELECT pk0000, pk0001, pk0002, ck0000, ck0001, static0000, static0001, static0002, static0003, static0004, regular0000, regular0001, regular0002, regular0003, writetime(static0000), writetime(static0001), writetime(static0002), writetime(static0003), writetime(static0004), writetime(regular0000), writetime(regular0001), writetime(regular0002), writetime(regular0003) FROM harry.table0 WHERE pk0000 = ? AND pk0001 = ? AND pk0002 = ? AND ck0000 <= ? ORDER BY ck0000 DESC, ck0001 DESC;', bindings=3765612582L,"ZHHyABdiABdiABdiABdiABdipWCSILEA1559158177179147197102174730239201982532068819411238236721811056853146156244879282122210020712931196418418619718113325123012386233","ZHHyABdiABdiABdiABdiABdiDBYfVZUm6240255198119168184222211241245114943702012482521291861632342511581252100835918786203192611772442472254501722156190381241094579",781480019256L}
```

Seems like there are rows returned, that the tool doesn't expect

**Comment by roydahan** [comment]
Well that's new :)
What scylla version did you use?

Let's merge and see what tests says in 5.0 and 2022.1.

BTW, why we don't want to backport it?

File: test-cases/longevity/longevity-harry-2h.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ instance_type_db: 'i3.large'
 
 aws_root_disk_size_loader: 80 # enlarge loader disk, cause of cassandra-harry operation.log that can't be disabled
 
-nemesis_class_name: 'SisyphusMonkey'
+nemesis_class_name: 'NoOpMonkey'
```
Comment (roydahan) [inline test-cases/longevity/longevity-harry-2h.yaml:12]: Is it for your testing?

Comment (fruch) [reply test-cases/longevity/longevity-harry-2h.yaml:12]: I'm not sure the this is ready for being run with a nemesis yet, we have issue with crashing see https://github.com/scylladb/scylla/issues/10553

**Comment by fruch** [comment]
> Well that's new :) What scylla version did you use?

I'm run it on master:latest, and run into this issue: https://github.com/scylladb/scylla/issues/10553

> 
> Let's merge and see what tests says in 5.0 and 2022.1.

I'll say lets merge without nemesis, and after we confirm it's working with 5.0 or 2022.1, we'll add a task to check with nemesis

> 
> BTW, why we don't want to backport it?

I wasn't sure it's ready to be test on release yet. but if you want we can.

**Comment by fruch** [comment]
@roydahan test was passing o.k. on 5.0.rc4, I'll merge it as is. I'll add a task for trying it with nemesis


Restore Monitor Stack command: `$ hydra investigate show-monitor 566c3e60-e647-46ec-b6d1-4a57f7e84b93`
Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?sct_branch=fix_cassandra_harry_install&test_id=566c3e60-e647-46ec-b6d1-4a57f7e84b93)
Show all stored logs command: `$ hydra investigate show-logs 566c3e60-e647-46ec-b6d1-4a57f7e84b93`

Test id: `566c3e60-e647-46ec-b6d1-4a57f7e84b93`


Logs:
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_145739/grafana-screenshot-longevity-harry-2h-test-scylla-per-server-metrics-nemesis-20220512_145859-longevity-harry-2h-fix-cass-monitor-node-566c3e60-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_145739/grafana-screenshot-longevity-harry-2h-test-scylla-per-server-metrics-nemesis-20220512_145859-longevity-harry-2h-fix-cass-monitor-node-566c3e60-1.png)
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_145739/grafana-screenshot-overview-20220512_145740-longevity-harry-2h-fix-cass-monitor-node-566c3e60-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_145739/grafana-screenshot-overview-20220512_145740-longevity-harry-2h-fix-cass-monitor-node-566c3e60-1.png)
db-cluster - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/db-cluster-566c3e60.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/db-cluster-566c3e60.tar.gz)
loader-set - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/loader-set-566c3e60.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/loader-set-566c3e60.tar.gz)
monitor-set - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/monitor-set-566c3e60.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/monitor-set-566c3e60.tar.gz)
sct - [https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/sct-runner-566c3e60.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/566c3e60-e647-46ec-b6d1-4a57f7e84b93/20220512_150707/sct-runner-566c3e60.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-harry-2h-test/8/)

image

**Comment by fgelcer** [comment]
we have a harry test for 5.0... should we backport this one to have it running for 5.0 and 2022.1?

**Comment by fruch** [comment]
> we have a harry test for 5.0... should we backport this one to have it running for 5.0 and 2022.1?

I did run it once on 5.0 (on staging) and it worked o.k.
I don't know if we'll learn much more running it again on 2022.1, I think it can wait for the next releases.

**Comment by roydahan** [comment]
We will run it on 5.0 and 2022.1 regardless of this fix.
AFAIU, it should still run even without this fix.

**Comment by fruch** [comment]
> We will run it on 5.0 and 2022.1 regardless of this fix. AFAIU, it should still run even without this fix.

I think it won't run, it was pointing the wrong repo, i.e. it wasn't pointing my fork anymore, someone "fixed" it to point the upstream.

**Comment by fruch** [comment]
@roydahan  I can backport it with the disable nemesis part, and let someone "shave" on it in 5.0/2022.1 ?

**Comment by roydahan** [comment]
No, I want to run it first on 5.0 with nemesis and see how it goes.

**Comment by fgelcer** [comment]
but anyway it still depends on part of this PR to be backported

**Pull request merged**